doc/s3ip_sysfs/s3ip_sysfs_hld.md

```diff
@@ -0,0 +1,339 @@
+
+The S3IP sysfs specification defines a unified interface to access peripheral hardware on devices from different vendors, making it easier for SONiC to support different devices and platforms.
+
+**Requirements for S3IP sysfs:**
+1. Should be able to work with the existing platform APIs
+2. The hierarchy of sysfs path structure should be defined in a specification.
+3. The sysfs path in the specification should be defined clearly, thus, platform API and test cases can be reused.
+4. The sysfs directory structure can be generated automatically through the framework.
+5. Support reuse of the old driver sysfs paths without redeveloping new drivers.
+
+**S3IP background Introduction**
+
+Simplified Switch System Integration Program, aka [S3IP](http://www.s3ip.org/), is the subproject of the Open Data Center Committee, aka [ODCC](http://www.opendatacenter.cn/introduction-en.html). S3IP is a joint initiative of six Internet companies in China, including Baidu, Alibaba, Tencent, Meituan, JD.com, and Kuaishou. Focusing on the network field, S3IP aims to solve the common problems in the integration process of the upper layer software and the lower layer equipment and establish a standardized ecology. The standardized design is more friendly to equipment manufacturers and enterprise users (such as the Internet), easier to integrate, more efficient to solve problems with the help of ecological forces, and faster to implement new technologies in the ecosystem.
+
+S3IP sysfs contains two sections:
+
+- sysfs specification: specifies the sysfs directory structure and file content
+- sysfs framework: generate the sysfs directory structure that conforming to the specification.
+
+### 2. S3IP sysfs specification
+#### 2.1 S3IP sysfs specification scenario
+
+Figure2-1 where S3IP sysfs is used.
+
+![scenario](/images/s3ip_sysfs/s3ip_sysfs_scenario.svg)
+
+The S3IP sysfs specification is represented as an organized sysfs directory structure on white-box devices. Device management software and debugging tools need to access hardware through this interface.
+
+Both vendors and users should comply with the S3IP sysfs specification. Device vendors focus on the specification implementation, while users verify the usability of the device against this specification. Vendors should provide software access to every sensor capable of being read. For any available sensor, vendors shall provide S3IP sysfs access.
+
+Figure2-2 S3IP sysfs diretory demo
+
+![demo](/images/s3ip_sysfs/s3ip_sysfs_demo.png)
+
+#### 2.2 Design of the S3ip sysfs specification
+Design Principles
+ - Unified, specificationized behavior, consistent experience among all SONiC devices, regardless of underlying platform; 

```

Comment:
"specificationized" is not a real word, I suggest "well-specified"

Reply:
ok，accept

Reply:
already updated